### PR TITLE
Migrate UserAccount helpers to UserAccountManager

### PR DESCRIPTION
**Pull request by ezaquarii**
This change migrate remaining AccountUtils static helpers
to UserAccountManager.

It contains no functional changes - static helpers code
is just moved into manager class and all clients are
supplied with manager as a dependency.

Signed-off-by: Chris Narkiewicz <hello@ezaquarii.com>

### CODE REVIEW

**Comment by ezaquarii** [comment]
xref #4046 3884

**Comment by ezaquarii** [comment]
Blocked by #4042

File: src/main/java/com/owncloud/android/datamodel/ArbitraryDataProvider.java [hunk 1]
```diff
@@ -156,7 +156,7 @@ public Integer getIntegerValue(Account account, String key) {
      * @return string if value found or empty string
      */
     public String getValue(Account account, String key) {
-        return getValue(account.name, key);
+        return account != null ? getValue(account.name, key) : null;
```
Comment (tobiasKaminsky) [inline src/main/java/com/owncloud/android/datamodel/ArbitraryDataProvider.java]: This mismatches to java doc and changes behaviour.
Can you return "" (empty string) and add @NonNull to it?

Comment (ezaquarii) [reply src/main/java/com/owncloud/android/datamodel/ArbitraryDataProvider.java]: Added `@NonNull` and made it non-null as well.

**Comment by ezaquarii** [comment]
@tobiasKaminsky @AndyScherzinger "Codacy was unable to analyse your pull request." WAT?

Sadly, no button to restart it.

**Comment by AndyScherzinger** [comment]
I'd say we ignore it. Just checked and also didn't find any way to trigger it 😔

**Comment by tobiasKaminsky** [comment]
I'll do a rebase to see if codacy will work then.

**Comment by ezaquarii** [comment]
@tobiasKaminsky 
```
+ ./gradlew createGplayDebugCoverageReport || scripts/uploadReport.sh $LOG_USERNAME $LOG_PASSWORD $DRONE_BUILD_NUMBER "IT" $DRONE_PULL_REQUEST $GIT_USERNAME $GIT_TOKEN
FAILURE: Build failed with an exception. 
Task 'createGplayDebugCoverageReport' not found in root project 'src'.
--

Then it tried to upload something for 30 minutes. Apart from this, looks ok.

**Comment by ezaquarii** [comment]
@tobiasKaminsky Can we merge it?

**Comment by tobiasKaminsky** [comment]
Thanks again for your awesome work!

**Pull request merged**